File: src/server/dekart/mime_test.go
package dekart

import "testing"

func TestGetFileExtensionFromMime(t *testing.T) {
	cases := map[string]string{
		"text/csv":                       "csv",
		"application/geo+json":           "geojson",
		"application/vnd.apache.parquet": "parquet",
		"application/octet-stream":       "parquet",
		"":                               "",
	}
	for mime, expected := range cases {
		if got := getFileExtensionFromMime(mime); got != expected {
			t.Fatalf("mime %q => %q, want %q", mime, got, expected)
		}
	}
}

func TestResolveUploadMimeType(t *testing.T) {
	cases := []struct {
		name     string
		mimeType string
		expected string
	}{
		{"sample.geojson", "application/octet-stream", "application/geo+json"},
		{"sample.parquet", "application/octet-stream", "application/vnd.apache.parquet"},
		{"sample.bin", "application/octet-stream", "application/octet-stream"},
		{"sample.csv", "application/geo+json", "application/geo+json"},
	}
	for _, testCase := range cases {
		if got := resolveUploadMimeType(testCase.name, testCase.mimeType); got != testCase.expected {
			t.Fatalf("resolveUploadMimeType(%q, %q) => %q, want %q", testCase.name, testCase.mimeType, got, testCase.expected)
		}
	}
}

func TestGetContentTypeFromExtensionCentral(t *testing.T) {
	cases := map[string]string{
		"csv":     "text/csv",
		"geojson": "application/geo+json",
		"parquet": "application/vnd.apache.parquet",
		"json":    "application/json",
		"":        "text/csv",
		"unknown": "text/csv",
	}
	for ext, expected := range cases {
		if got := getContentTypeFromExtensionCentral(ext); got != expected {
			t.Fatalf("ext %q => %q, want %q", ext, got, expected)
		}
	}
}

File: src/client/lib/mime.test.js
import { describe, it, expect } from 'vitest'
import { mimeFromExtension, extensionFromMime, inferMimeFromName, contentTypeFromExtension } from './mime'

describe('mime helpers (client)', () => {
  it('maps extension to mime', () => {
    expect(mimeFromExtension('csv')).toBe('text/csv')
    expect(mimeFromExtension('.geojson')).toBe('application/geo+json')
    expect(mimeFromExtension('parquet')).toBe('application/vnd.apache.parquet')
    expect(mimeFromExtension('json')).toBe('application/json')
  })

  it('maps mime to extension', () => {
    expect(extensionFromMime('text/csv')).toBe('csv')
    expect(extensionFromMime('application/geo+json')).toBe('geojson')
    expect(extensionFromMime('application/vnd.apache.parquet')).toBe('parquet')
    expect(extensionFromMime('application/octet-stream')).toBe('parquet')
    expect(extensionFromMime('application/json')).toBe('json')
  })

  it('infers mime from filename', () => {
    expect(inferMimeFromName('file.CSV')).toBe('text/csv')
    expect(inferMimeFromName('data.geojson')).toBe('application/geo+json')
    expect(inferMimeFromName('new-test.parquet')).toBe('application/vnd.apache.parquet')
    expect(inferMimeFromName('unknown.bin')).toBe('')
  })

  it('contentTypeFromExtension defaults to csv', () => {
    expect(contentTypeFromExtension('csv')).toBe('text/csv')
    expect(contentTypeFromExtension('geojson')).toBe('application/geo+json')
    expect(contentTypeFromExtension('parquet')).toBe('application/vnd.apache.parquet')
    expect(contentTypeFromExtension('json')).toBe('application/json')
    expect(contentTypeFromExtension('unknown')).toBe('text/csv')
  })
})
